cypress/e2e/CMS/$Article/index.js

```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
     cy.get("div[class='public-use']").should('not.exist');
 });
 
+And('user selects {string} content item', (dropDown) => {
+    cy.get(`input[value='${dropDown}']`).click({ force: true });
+});
 
+And('user clicks on {string} link in the {string} text area', (title, cartOption) => {
+    cy.get(`div.paragraph-type-title:contains('${cartOption}')`).parent().parent().find(`span:contains('${title}')`).parent().click();
+});
 
+And('user clicks on {string} button item', (title) => {
+    cy.get(`input[value="${title}"]`).click({ force: true })
+});
 
+And('user selects {string} item from main page content', (title) => {
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find(`td:contains("${title}")`).parent().find('input').check();
+});
 
+Then('the promo image is matching the earlier selected image', () => {
+    function extractImgName(imageFullName) {
+        let str = '';
+        for (let i = 0; i < imageFullName.length; i++) {
+            let stringToRemove = '';
+            if (imageFullName.charAt(i) === '/') {
+                for (let j = i; j < imageFullName.length; j++) {
+                    stringToRemove += imageFullName.charAt(j);
+                    if (imageFullName.charAt(i + 1) === '/') {
+                        return;
+                    }
+                }
+                str = stringToRemove;
+            }
+        }
+        return str;
+    }
+    const expectedSrc = (imageSrc1.replace(/\?itok=[\S]+/, '')).replace(/^(.*?)\/public/, '');
+    const extractedImageName = extractImgName(expectedSrc).replace(/\.jpg|\.jpeg|\.png/, '')
 
+    cy.get('div.feature-card').find('img').then($el => {
+        const source = $el[0].getAttribute('src');
+        const actSrc = source.replace(/\?itok=[\S]+/, '').replace(/^(.*?)\/public/, '')
+        expect(actSrc).to.include(extractedImageName.replaceAll('_', '-').replace('article', ''))
+    })
+});
 
+And('user selects the {string} content item', (dropDown) => {
```

Comment:
this step is duplicated 8 times - the code and locators are the same, no reason to copy paste it so many times, put it in the commons

Reply:
done